### DISCUSSION TITLE
Added Pushy 0.14.2 version, it sending 2 notifications to device one for one event.

### DISCUSSION THREAD

**Discussion by ramyabugowda-android** [relayrides/pushy#883] [Q&A]
When trigger a event from server two send one notification to iOS client, it sends 2 notifications for the same event, I have even handled the removal of invalid device tokens as mentioned below, please suggest if something is wrong.

```java
private void handlePushNotificationResponsePerToken(final PushNotificationResponse<SimpleApnsPushNotification> pushNotificationResponse ) {
    final String deviceToken = pushNotificationResponse.getPushNotification().getToken();
    if (pushNotificationResponse.isAccepted()) {
        // log information
    
    }
    else {
        if ((pushNotificationResponse.getTokenInvalidationTimestamp() != null) || ("BadDeviceToken".equals(rejectReason))) {
            pushNotificationResponse.getTokenInvalidationTimestamp().ifPresent(timestamp -> {
                Date deviceTokenTimeStamp = Date.from(timestamp);
                inactiveDeviceTokens.put(deviceToken,deviceTokenTimeStamp);
            });
        }
    }
}
```

**Comment by jchambers** [comment]
Hello! This sounds more like a request for general support than a bug report or feature request, so I'm going to turn this into a discussion. I'll follow up on the discussion side of things.

**Comment by jchambers** [comment]
It looks like the code you've shared is the part that handles the response; if you're worried about double-sending notifications, I think we'd need to see the code that actually sends the notifications.

Additionally, if you can [capture some HTTP/2 frame logs](https://pushy-apns.org/apidocs/0.14/com/eatthepath/pushy/apns/ApnsClientBuilder.html#setFrameLogger(io.netty.handler.codec.http2.Http2FrameLogger)), that would go a long way to helping us understand what might be happening in your case.
